staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -104,6 +104,19 @@ func NewEventFromRequest(req *http.Request, level auditinternal.Level, attribs a
 	return ev, nil
 }
 
+// UpdateEventAttribs updates authorizer attributes according to attribs
+func UpdateEventAttribs(ev *auditinternal.Event, attribs authorizer.Attributes) {
+	if user := attribs.GetUser(); user != nil {
+		ev.User.Username = user.GetName()
```

Comment:
Don't copy code, I'd expect impersonated user to hold the output of the impersonation filter, what you are calling the effective user